### PR TITLE
Internal: Infra - Allow to locate the editor in any selector [ED-2550]

### PR DESCRIPTION
**Pull request by iNewLegend**

### CODE REVIEW

**Review by matipojo — CHANGES_REQUESTED** [review]

File: assets/dev/js/editor/components/documents/commands/internal/attach-preview.js [hunk 1]
```diff
@@ -34,7 +42,7 @@ export class AttachPreview extends $e.modules.CommandInternalBase {
 				return resolve();
 			}
 
-			document.$element = elementor.$previewContents.find( '.elementor-' + document.id );
+			document.$element = selector || elementor.$previewContents.find( '.elementor-' + document.id );
```
Comment (matipojo) [inline assets/dev/js/editor/components/documents/commands/internal/attach-preview.js:45]:
```suggestion
			document.$element = jQuery( selector ) || elementor.$previewContents.find( '.elementor-' + document.id );
```

File: assets/dev/js/editor/components/documents/commands/switch.js [hunk 2]
```diff
@@ -4,7 +4,7 @@ export class Switch extends $e.modules.CommandBase {
 	}
 
 	apply( args ) {
-		const { id, mode, onClose } = args;
+		const { id, mode, onClose, selector } = args;
```
Comment (matipojo) [inline assets/dev/js/editor/components/documents/commands/switch.js:7]: JSLint

**Review by matipojo — CHANGES_REQUESTED** [review]

File: assets/dev/js/editor/components/documents/commands/internal/attach-preview.js (hunk 1 above)
Comment (iNewLegend) [reply assets/dev/js/editor/components/documents/commands/internal/attach-preview.js]: Handled.

File: assets/dev/js/editor/components/documents/commands/switch.js (hunk 2 above)
Comment (iNewLegend) [reply assets/dev/js/editor/components/documents/commands/switch.js]: Handled.

File: assets/dev/js/editor/components/documents/commands/internal/attach-preview.js [hunk 3]
```diff
@@ -1,12 +1,18 @@
 export class AttachPreview extends $e.modules.CommandInternalBase {
-	apply() {
+	validateArgs( args = {} ) {
+		if ( args.selector ) {
+			this.requireArgumentConstructor( 'selector', jQuery );
+		}
+	}
```
[the remark this answers is not in the document]
Comment (iNewLegend) [reply assets/dev/js/editor/components/documents/commands/internal/attach-preview.js]: Handled.

File: assets/dev/js/editor/components/documents/commands/internal/attach-preview.js [hunk 4]
```diff
@@ -23,7 +33,9 @@ export class AttachPreview extends $e.modules.CommandInternalBase {
 		} );
 	}
 
-	attachDocumentToPreview( document ) {
+	attachDocumentToPreview( document, args ) {
+		const { selector } = args;
```
Comment (matipojo) [inline assets/dev/js/editor/components/documents/commands/internal/attach-preview.js:37]:
```suggestion
		let { selector } = args;
```

File: assets/dev/js/editor/components/documents/commands/internal/attach-preview.js [hunk 5]
```diff
@@ -34,7 +46,11 @@ export class AttachPreview extends $e.modules.CommandInternalBase {
 				return resolve();
 			}
 
-			document.$element = elementor.$previewContents.find( '.elementor-' + document.id );
+			if ( selector?.length ) {
+				document.$element = elementor.$previewContents.find( selector );
+			} else {
+				document.$element = elementor.$previewContents.find( '.elementor-' + document.id );
+			}
```
Comment (matipojo) [inline assets/dev/js/editor/components/documents/commands/internal/attach-preview.js:53]:
```suggestion
			selector ||= '.elementor-' + document.id;
			document.$element = elementor.$previewContents.find(  );
			
```

**Review by matipojo — CHANGES_REQUESTED** [review]

File: assets/dev/js/editor/components/documents/commands/internal/attach-preview.js (hunk 5 above)
Comment (iNewLegend) [reply assets/dev/js/editor/components/documents/commands/internal/attach-preview.js:53]: Parsing error: This experimental syntax requires enabling the parser plugin: 'logicalAssignment'

Comment (matipojo) [reply assets/dev/js/editor/components/documents/commands/internal/attach-preview.js:53]: Just add it

Comment (iNewLegend) [reply assets/dev/js/editor/components/documents/commands/internal/attach-preview.js]: Handled other way:)

File: tests/playwright/pages/wp-admin-page.js [hunk 6]
```diff
@@ -62,6 +62,7 @@ module.exports = class WpAdminPage extends BasePage {
 
 	async waitForPanel() {
 		await this.page.waitForSelector( '.elementor-panel-loading', { state: 'detached' } );
+		await this.page.waitForTimeout( 5000 );
```
Comment (matipojo) [inline tests/playwright/pages/wp-admin-page.js:65]: Why? Try to wait for a specific event.

**Review by matipojo — CHANGES_REQUESTED** [review]

File: tests/playwright/pages/wp-admin-page.js (hunk 6 above)
Comment (iNewLegend) [reply tests/playwright/pages/wp-admin-page.js:65]:
```suggestion
```

**Pull request merged**